lib/Dialect/TritonGPU/Transforms/WarpSpecialization/PartitionLoops.cpp

```diff
@@ -16,103 +19,331 @@ using namespace mlir;
 using namespace triton;
 using namespace triton::gpu;
 
-//===----------------------------------------------------------------------===//
-// slicePartition
-//===----------------------------------------------------------------------===//
+namespace {
+
+struct WarpGroupBuilder : public OpBuilder {
+  using OpBuilder::OpBuilder;
+
+  static WarpGroupBuilder atBlockEnd(Block *block, size_t partitionId) {
+    WarpGroupBuilder builder(block, block->end(), nullptr);
+    builder.mapping = IRMapping();
+    builder.partitionId = partitionId;
+    return builder;
+  }
+
+  IRMapping mapping;
+  size_t partitionId;
+};
+
+// This is computed per loop and partition
+enum class LoopVarCategory {
+  Unused,
```

Comment:
can you add some comments describing the situations in which a Value is put into each category?

Reply:
Added more descriptions